### PR TITLE
Feature Implementation: Invitation Flow for New and Existing Users with Email Notifications

### PR DESCRIPTION
**Pull request by HectorAgudelo**
This pull request introduces a new feature that enhances the user invitation flow for our application. Key changes include:

- Differentiated Email Handling: The Lambda function now dynamically generates email invitations based on whether the guest is a new or existing user. New users receive registration links, while existing users receive login links.

- Customizable Email Templates: The email content and design have been improved to provide a more welcoming and informative experience for invited users. (missing or needs customization).

- Invitation Status Tracking: A more robust system for tracking invitation statuses has been implemented, including handling for revoked, expired, unexpected status, and not found scenarios.

- Error Handling and Rollback: The Lambda function includes improved error handling with transaction rollbacks to ensure data integrity in case of failures.

- Frontend Integration: Added a modal to the frontend to display relevant messages to the user, based on the invitation status, so they are aware of any issues with their invitation.

Technical Details:

- Lambda Function (handler): The primary logic for invitation creation, email sending, and status management is handled here.

- Database Interaction: The function interacts with the database to store and retrieve invitation data.

- AWS SES Integration: Amazon Simple Email Service (SES) is used to send the customized email invitations.

- Frontend Components: The frontend was modified to display a modal to the user with relevant information, based on the invitation status.

Additional Notes:

- The code has been tested thoroughly to ensure it functions correctly for both new and existing users.

- Please review the code changes and provide feedback.

Files Changed:

- amplify/backend/auth/athletifiAuth/cli-inputs.json
- amplify/backend/backend-config.json
- src/app/(auth)/profile/page.tsx
- src/app/utils/auth/handlePostSignIn.ts
- src/components/auth/LoggedOut.tsx
- src/components/common/AlertModal.tsx
- src/states/userStore.ts
- src/types/AlertModalType.ts

Functional Testing:

New User Invitation Flow:

- Create a test email address that doesn't exist in your user database. (Use the referralInvite lambda function to do this)
- Use the application to send an invitation to this test email. (Use the referralInvite lambda function to do this)
- Verify that the correct email is received with a registration link (not a login link).
- Follow the registration link and complete the registration process. URL example with invitation Id: 
  http://localhost:3000/register?invite_id=ba71d78e-2b20-4bef-aa4e-6b8153fd031b
- Ensure that the new user is successfully created in the database and linked to the invitation.
- Verify the modal displays the correct invitation message. (no modal should be shown if everything runs well) check the lambda function writeUserSignUp starting on line 135.

Existing User Invitation Flow:

- Use an existing user's email address in your application to send an invitation.
- Confirm that the existing user receives an email with a login link.
- Follow the login link and ensure the user is directed to the appropriate page within the application.
- Verify the modal displays the correct invitation message.

Error Scenarios:

- Revoked Invitation: Revoke an invitation before it's accepted and verify the appropriate error message is displayed in the modal.
- Expired Invitation: Let an invitation expire and confirm the correct error message is shown in the modal.
- Invalid Invitation: Attempt to access the application with an invalid or non-existent invitation ID and verify the error message in the modal.
- Unexpected Status: Introduce an unexpected status in your backend (e.g., by modifying the database directly) and verify that the general error message is displayed in the modal.

### CODE REVIEW

**Review by qisforq — CHANGES_REQUESTED** [review]
@HectorAgudelo just finished reviewing and testing!

Just a couple things:

1. I did not receive a warning when the invite was already accepted. However I did receive a warning for expired and revoked.
2. Instead of the status being '1 ALREADY_ACCEPTED', can we change this back to 'ALREADY_ACCEPTED'?

**Comment by HectorAgudelo** [comment]
> @HectorAgudelo just finished reviewing and testing!
> 
> Just a couple things:
> 
> 1. I did not receive a warning when the invite was already accepted. However I did receive a warning for expired and revoked.
> 2. Instead of the status being '1 ALREADY_ACCEPTED', can we change this back to 'ALREADY_ACCEPTED'?

Fixed!

**Comment by qisforq** [comment]
> > @HectorAgudelo just finished reviewing and testing!
> > Just a couple things:
> > 
> > 1. I did not receive a warning when the invite was already accepted. However I did receive a warning for expired and revoked.
> > 2. Instead of the status being '1 ALREADY_ACCEPTED', can we change this back to 'ALREADY_ACCEPTED'?
> 
> Fixed!

Awesome thanks @HectorAgudelo! I do not see it here yet, can you push your changes?

File: src/app/(auth)/profile/page.tsx [hunk 1]
```diff
@@ -71,6 +74,54 @@ const variants = {
...
+  console.log('inviteData', inviteData);
+  // console.log('inviteStatus', inviteStatus);
+
+  useEffect(() => {
+    const hasShownModal = localStorage.getItem('hasShownModal');
+    if (
+      !hasShownModal &&
+      inviteData &&
+      inviteData.invitation.invite_id &&
+      // inviteData.invitation.invite_status !== 'ALREADY_ACCEPTED' &&
```
Comment (qisforq) [inline src/app/(auth)/profile/page.tsx:88]: @HectorAgudelo is this supposed to be commented out?

Comment (qisforq) [reply src/app/(auth)/profile/page.tsx:88]: After we discussed, it looks like this line was supposed to be taken out so this is good

**Review by qisforq — APPROVED** [review]

**Pull request merged**